functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py

```diff
@@ -366,16 +366,40 @@ def test011_restart_vm_after_migration(self):
 
         self.lg('%s ENDED' % self._testID)
 
-    @unittest.skip('Not Implemented')
     def test012_check_cloned_vm(self):
-        """ OVC-000
+        """ OVC-029
         *Test case for checking cloned VM ip, portforwards and credentials*
-
         **Test Scenario:**
-
-        #. Create (VM1).
+        #. Create (VM1), should succeed.
+        #. Create portforward to ssh port for (VM1).
+        #. Stop (VM1), should succeed.
         #. Clone VM1 as (VM2_C), should succeed.
         #. Make sure VM2_C got a new ip.
-        #. Check that VM2 got different credentials from VM1.
         #. Make sure no portforwards have been created.
         """
+        self.lg('Create (VM1), should succeed')
+        machineId = self.cloudapi_create_machine(self.cloudspace_id)
+        self.wait_for_machine_to_get_ip(machineId)
```

Comment:
check on the ip before proceeding .. in this case u may return False in wait_for_machine_to_get_ip

Reply:
done